app/src/display/main.c

```diff
@@ -85,33 +85,41 @@ static void stop_display_updates() {
 
 int zmk_display_is_initialized() { return initialized; }
 
-int zmk_display_init() {
+void initialize_display(struct k_work *work) {
     LOG_DBG("");
 
     display = device_get_binding(ZMK_DISPLAY_NAME);
     if (display == NULL) {
         LOG_ERR("Failed to find display device");
-        return -EINVAL;
+        return;
     }
 
-#if IS_ENABLED(CONFIG_ZMK_DISPLAY_WORK_QUEUE_DEDICATED)
-    k_work_q_start(&display_work_q, display_work_stack_area,
-                   K_THREAD_STACK_SIZEOF(display_work_stack_area),
-                   CONFIG_ZMK_DISPLAY_DEDICATED_THREAD_PRIORITY);
-#endif
-
     screen = zmk_display_status_screen();
 
     if (screen == NULL) {
         LOG_ERR("No status screen provided");
-        return 0;
+        return;
     }
 
     lv_scr_load(screen);
 
     start_display_updates();
 
     initialized = true;
+}
+
+K_WORK_DEFINE(init_work, initialize_display);
+
+int zmk_display_init() {
+    LOG_DBG("");
```

Comment:
What are the blank debug messages used for?

Reply:
Silliness! Removed.